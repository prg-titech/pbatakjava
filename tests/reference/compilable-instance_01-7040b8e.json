{
    "basename": "compilable-instance_01-7040b8e",
    "cmd": "{infile}",
    "infile": "tests/instance_01/build.sh",
    "infile_hash": "9bcf3e27e560ccd948871dca814364ccb6a295f6b484f869005d7a10",
    "stdout": "compilable-instance_01-7040b8e.stdout",
    "stdout_hash": "aa603e78356ef629aabbdce6ed09bf1eec025b903b79af25af8c84f2",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}