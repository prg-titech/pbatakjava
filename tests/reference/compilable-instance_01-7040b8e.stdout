
public interface Point {
  public int get__x();

  public int set__x(int toAssign);

  public int get__y();

  public int set__y(int toAssign);
}

public interface Point_Factory {
  public Point make(int x, int y);
}

public class Point_v1 implements Point {
  public int x;

  public int y;

  public Point_v1(int x, int y) {
    this.x = x;
    this.y = y;
  }

  public int get__x() {
    return this.x;
  }

  public int set__x(int toAssign) {
    return this.x = toAssign;
  }

  public int get__y() {
    return this.y;
  }

  public int set__y(int toAssign) {
    return this.y = toAssign;
  }
}

public class Point_v1_Factory implements Point_Factory {
  public Point make(int x, int y) {
    return new Point_v1(x, y);
  }
}

public class Point_v2 implements Point {
  public int x;

  public int y;

  public Point_v2(int x, int y) {
    this.x = x;
    this.y = y;
  }

  public int get__x() {
    return this.x;
  }

  public int set__x(int toAssign) {
    return this.x = toAssign;
  }

  public int get__y() {
    return this.y;
  }

  public int set__y(int toAssign) {
    return this.y = toAssign;
  }
}

public class Point_v2_Factory implements Point_Factory {
  public Point make(int x, int y) {
    return new Point_v2(x, y);
  }
}

public interface Test {
}

public interface Test_Factory {
}

public class Test_v1 implements Test {
  public static void main(String[] args) {
    Point p = new Point_v1(1, 1);
  }
}

public class Test_v1_Factory implements Test_Factory {
}
